app/src/main/java/com/example/pix/chat/adapters/MessageAdapter.java

```diff
@@ -82,17 +91,64 @@ public void bind(int position, List<Message> messages) {
                 this.text.setVisibility(View.VISIBLE);
                 this.text.setText("" + message.getText());
             }
 
 
             ParseFile pic = message.getPic();
-            if (pic != null) {
+            // Only show pic attachment if not a Snap pic
+            if (pic != null && !message.isSnap()) {
                 this.contentPic.setVisibility(View.VISIBLE);
-                Glide.with(this.contentPics.getContext()).load(pic.getUrl()).into(this.contentPic);
+                Glide.with(context).load(pic.getUrl()).into(this.contentPic);
+            }
+
+
+            // If this is a Snap picture...
+            if (message.isSnap()) {
+                // Show the snap layout
+                this.openSnap.setVisibility(View.VISIBLE);
+
+                // If this User sent it, they can't open the Snap
+                if (message.getFrom().getObjectId().equals(ParseUser.getCurrentUser().getObjectId())) {
+                    this.snapText.setText("Delivered");
+                } else {
+                    // If this User received the Snap picture, if they click the layout...
+                    this.openSnap.setOnClickListener(unusedView -> {
+                        // Set up the picture as an ImageView that lives in ChatFragment
+                        // This ImageView is placed above the who Chat with a friend with visibility Gone,
+                        // so we change it to visible and it will cover the screen
+                        ParseFile picture = message.getPic();
+                        Glide.with(context).load(picture.getUrl()).into(snapContainer);
+                        snapContainer.setVisibility(View.VISIBLE);
+
+                        // If we click this new Imageview...
+                        snapContainer.setOnClickListener(unusedView1 -> {
+                            // Get rid of the Snap picture and hide the big ImageView
+                            snapContainer.setImageResource(0);
+                            snapContainer.setVisibility(View.GONE);
+
+
+                            Drawable buttonDrawable = this.snapIndicator.getBackground();
+                            buttonDrawable = DrawableCompat.wrap(buttonDrawable);
+                            // Indicate to User that we have opened the Snap by changing indicator color
+                            DrawableCompat.setTint(buttonDrawable, Color.WHITE);
+                            this.snapIndicator.setBackground(buttonDrawable);
+
+                            // Delete Snap
+                            message.deleteInBackground();
+                            this.snapText.setText("Opened");
+
+                            // No more interaction with this empty Snap
+                            snapContainer.setOnClickListener(null);
+                            this.openSnap.setOnClickListener(null);
+                        });
+                    });
+                }
```

Comment:
The change to this file looks similar to the Snap PR. Is this PR intended to depend on the other one?

Reply:
I updated this branch by merging the Snaps branch into it but it doesn't depend on it.